docs/sources/v0.51.x/using-k6-browser/migrating-to-k6-v0-52.md

```diff
@@ -0,0 +1,215 @@
+        },
+      },
+    },
+  },
+  thresholds: {
+    checks: ['rate==1.0'],
+  },
+};
+
+export default async function () {
+  const context = await browser.newContext();
+  const page = await context.newPage();
+
+  try {
+    // Navigate to the website under test.
+    await page.goto('https://test.k6.io/', { waitUntil: 'networkidle' });
+    await Promise.all([
+      page.waitForNavigation(),
+      page.locator('a[href="/my_messages.php"]').click(),
+    ]);
+
+    // Check that the header reads "Unauthorized".
+    const h2 = page.locator('h2');
+    const headerOK = (await h2.textContent()) == 'Unauthorized';
+    check(headerOK, { header: headerOK });
+
+    // Take a screenshot of the page.
+    await page.screenshot({ path: 'screenshot.png' });
+  } finally {
+    await page.close();
+  }
+}
+```
+
+{{< /code >}}
+
+### Can I promise chain in v0.51?
+
+If you want to use promise chaining the newly migrated async APIs, You won't be able to do that until the migration. For example, you can't do this yet:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  await page.$('#text1').then((text) => text.type('hello world'));
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+Instead you could do this until k6 `v0.52` is released:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  const text = await page.$('#text1');
+  await text.type('hello world');
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+## Working with k6 check
+
+The k6 `check` API wasn't designed to work with async APIs. However there is a workaround. In your current scripts, you may use a `check` like so:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import { check } from 'k6';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  check(page, {
+    header: (p) => p.locator('h2').textContent() == 'Welcome, admin!',
+  });
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+Since the `locator.textContent` API is being migrated to async, and `check` doesn't work with async APIs, you will need to do something like this:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import { check } from 'k6';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  const h2 = page.locator('h2');
+  const headerOK = (await h2.textContent()) == 'Welcome, admin!';
+  check(headerOK, { header: headerOK });
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+## Groups
+
+A note on groups, they do not work with async APIs either, there is no workaround as of yet. Here's the [github issue](https://github.com/grafana/k6/issues/2728) that you can follow to keep up to date with relevant news on a groups API that works with async APIs.
```

Comment:
Is it worth mentioning this? It's likely that people have already tried this and realised that it doesn't work since a `group` might contain a `page.goto` or `page.click` API call, which are both async already.

Reply:
My two cents, I think it's worth adding it here just to be explicit that this is a limitation. It could save someone time versus discovering it on their own.